networkx/algorithms/traversal/breadth_first_search.py

```diff
@@ -67,25 +67,30 @@ def generic_bfs_edges(G, source, neighbors=None, depth_limit=None, sort_neighbor
     .. _PADS: http://www.ics.uci.edu/~eppstein/PADS/BFS.py
     .. _Depth-limited-search: https://en.wikipedia.org/wiki/Depth-limited_search
     """
-    if callable(sort_neighbors):
+    if neighbors is None:
+        neighbors = G.neighbors
+    if sort_neighbors is not None:
         _neighbors = neighbors
         neighbors = lambda node: iter(sort_neighbors(_neighbors(node)))
-
-    visited = {source}
     if depth_limit is None:
-        depth_limit = len(G)
-    queue = deque([(source, depth_limit, neighbors(source))])
-    while queue:
-        parent, depth_now, children = queue[0]
-        try:
-            child = next(children)
-            if child not in visited:
-                yield parent, child
-                visited.add(child)
-                if depth_now > 1:
-                    queue.append((child, depth_now - 1, neighbors(child)))
-        except StopIteration:
-            queue.popleft()
+        depth_limit = float("inf")
```

Comment:
What's the benefit to this? It seems like it could only lead to worse outcomes than len(G).

Reply:
It was more about consistency between this PR and the other I opened about bfs functions, but I think you are right, it doesn't matter much changing it or not and considering consistency intra-module I think your suggestions are better!